src/core_modules/capture-core/rules/RuleEngine/RuleEngine.js

```diff
@@ -0,0 +1,113 @@
+// @flow
+import { RuleEngineJs } from '@dhis2/rule-engine';
+import {
+    InputBuilder,
+    ValueProcessor,
+    getRulesEffectsProcessor,
+} from './helpers';
+import type {
+    OutputEffects,
+    RulesEngineInput,
+    IConvertInputRulesValue,
+    IConvertOutputRulesEffectsValue,
+    Flag,
+} from './types/ruleEngine.types';
+
+export class RuleEngine {
+    inputConverter: IConvertInputRulesValue;
+    outputConverter: IConvertOutputRulesEffectsValue;
+    valueProcessor: ValueProcessor;
+    userRoles: Array<string>;
+    flags: Flag;
+
+    constructor(
+        inputConverter: IConvertInputRulesValue,
+        outputConverter: IConvertOutputRulesEffectsValue,
+        flags?: Flag,
+    ) {
+        this.inputConverter = inputConverter;
+        this.outputConverter = outputConverter;
+        this.valueProcessor = new ValueProcessor(inputConverter);
+        this.flags = flags ?? {};
+    }
+
+    getProgramRuleEffects({
+        programRulesContainer,
+        currentEvent,
+        otherEvents,
+        dataElements,
+        trackedEntityAttributes,
+        selectedEntity,
+        selectedEnrollment,
+        selectedOrgUnit,
+        selectedUserRoles,
+        optionSets,
+    }: RulesEngineInput): OutputEffects {
+        if (!programRulesContainer.programRules ||
+            !selectedOrgUnit ||
+            !currentEvent && !selectedEnrollment) return [];
```

Comment:
Might work already, but easier to read with the parentheses 
```suggestion
        if (!programRulesContainer.programRules ||
            !selectedOrgUnit ||
            (!currentEvent && !selectedEnrollment)) return [];
```